### DISCUSSION TITLE
关于alist使用onlyoffice的配置问题

### DISCUSSION THREAD

**Discussion by hhy5562877** [AlistGo/alist#3255] [Q&A] [answered]
搭建好only office之后填进去Url 打开后显示的是welcome页面
已经搜索了相关问题， 但是填上之后显示
Cannot GET /view.html
请问一下应该怎么去配置alist+only office呢

**Comment by tingyue-wu** [comment]
同求配置

**Comment by chenjuefei** [comment, marked as answer]
因为没有view.html 需要你自己编写


```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>OnlyOffice Viewer</title>
</head>
 
<body>
    <div id="placeholder"></div>
    <script type="text/javascript" src="https://documentserver/web-apps/apps/api/documents/api.js"></script>
    <script>
        function getQueryParamValue(name) {
            const searchParams = new URLSearchParams(window.location.search);
            return searchParams.get(name);
        }
        
        const url = decodeURIComponent(getQueryParamValue("src"));
        const fileName = url.substring(url.lastIndexOf('/') + 1, url.lastIndexOf('?') != -1 ? url.lastIndexOf('?') : url.length);
        const fileExtension = fileName.split('.').pop();
        const docEditor = new DocsAPI.DocEditor("placeholder", {
            "document": {
                "fileType": fileExtension,
                "permissions": {
                    "edit": false,
                    "comment": true,
                    "download": true,
                    "print": true,
                    "fillForms": true,
                },
                "title": fileName,
                "url": url,
            },
            "editorConfig": {
                "lang": "zh-CN",
                "mode": "view",
            },
            "height": "1080px",
            "type": "desktop",
        });
    </script>
</body>
</html>
```
![image](https://user-images.githubusercontent.com/35056864/219602826-fe2a0e9f-3a4b-4e9e-b442-0ca2324b2977.png)

另：onlyoffice服务器，需要关闭JWT不然一直报“文档安全令牌未正确形成”。如果有大神研究下不用关闭也能用的话，那最好！如有BUG。欢迎指正！

![image](https://user-images.githubusercontent.com/35056864/219597958-6ba1822a-a082-488a-9fc5-a754fc96f302.png)

**Reply by tingyue-wu** [reply]
> view.html

太感谢了。那么能告知view.html 文件添加到哪里呢？

**Reply by chenjuefei** [reply]
任何一个能显示HTML地方都行。实在找不到的话，你也可以放在api.js的边上。访问就是“https://documentserver/web-apps/apps/api/documents/view.html”   终端路径  /var/www/onlyoffice/documentserver/web-apps/apps/api/documents  不过我没这么做。。。你自己研究下吧

**Reply by Westwall98** [reply]
把script的src改为api.js的地址，然后alist的框架中会显示无法访问文件，是文件流返回得不对吗？

**Reply by tingyue-wu** [reply]
> 任何一个能显示HTML地方都行。实在找不到的话，你也可以放在api.js的边上。访问就是“[https://documentserver/web-apps/apps/api/documents/view.html”](https://documentserver/web-apps/apps/api/documents/view.html%E2%80%9D) 终端路径 /var/www/onlyoffice/documentserver/web-apps/apps/api/documents 不过我没这么做。。。你自己研究下吧

已经把view.html文件一字未改创建好，并放在onlyoffice容器的/var/www/onlyoffice/documentserver/web-apps/apps/api/documents位置，然后重启了onlyoffice容器，但是alist依然提示Cannot GET /view.html

**Reply by tingyue-wu** [reply]
> 任何一个能显示HTML地方都行。实在找不到的话，你也可以放在api.js的边上。访问就是“[https://documentserver/web-apps/apps/api/documents/view.html”](https://documentserver/web-apps/apps/api/documents/view.html%E2%80%9D) 终端路径 /var/www/onlyoffice/documentserver/web-apps/apps/api/documents 不过我没这么做。。。你自己研究下吧

另外，我发现zfile这个列表工具是直接填写you_only_office_url不需要view.html也能预览的，不知道它后面匹配的是什么命令

**Reply by chenjuefei** [reply]
Cannot GET /view.html 说明访问不到这个页面。你得检查你网址路径 。 代码里 documentserver 要改成你自己的。

**Reply by tingyue-wu** [reply]
> 把script的src改为api.js的地址，然后alist的框架中会显示无法访问文件，是文件流返回得不对吗？

你弄好了吗，我现在view文件可以加载了，但是打开文档还是显示空白页面
![image](https://user-images.githubusercontent.com/62607946/219942293-94bf708d-9835-47fc-8c29-8bf2ccf1c26f.png)

**Reply by chenjuefei** [reply]
1. 首先搭好onlyoffice/documentserver后，浏览网址 `https://你的域名:端口/web-apps/apps/api/documents/api.js` 查看能否看到内容。如果没有说明没有搭建好onlyoffice或者路径错误。 我这里域名地址访问，内网IP地址后续会有问题，建议用域名。
2. 编辑好view.html 唯一要修改的api.js地址，改成你自己的。浏览网址 `https://你的域名:端口/路径/view.html` 如果显示 `The "document.fileType" parameter for the config object is invalid. Please correct it.` 表示正常。没有显示，或者显示`Cannot GET /view.html `回头检查。
3. 最后 Iframe 预览里 添加view.html路径 如图，后缀添加 `?src=$e_url`

**Reply by tingyue-wu** [reply]
> 1. 首先搭好onlyoffice/documentserver后，浏览网址 `https://你的域名:端口/web-apps/apps/api/documents/api.js` 查看能否看到内容。如果没有说明没有搭建好onlyoffice或者路径错误。 我这里域名地址访问，内网IP地址后续会有问题，建议用域名。
> 2. 编辑好view.html 唯一要修改的api.js地址，改成你自己的。浏览网址 `https://你的域名:端口/路径/view.html` 如果显示 `The "document.fileType" parameter for the config object is invalid. Please correct it.` 表示正常。没有显示，或者显示`Cannot GET /view.html `回头检查。
> 3. 最后 Iframe 预览里 添加view.html路径 如图，后缀添加 `?src=$e_url`

搞好了，原来view.html页面所在的那个容器需要重启才行，之前没有重启，所以页面显示空白。谢谢您的辛苦指点！！！

**Reply by tingyue-wu** [reply]
> 1. 首先搭好onlyoffice/documentserver后，浏览网址 `https://你的域名:端口/web-apps/apps/api/documents/api.js` 查看能否看到内容。如果没有说明没有搭建好onlyoffice或者路径错误。 我这里域名地址访问，内网IP地址后续会有问题，建议用域名。
> 2. 编辑好view.html 唯一要修改的api.js地址，改成你自己的。浏览网址 `https://你的域名:端口/路径/view.html` 如果显示 `The "document.fileType" parameter for the config object is invalid. Please correct it.` 表示正常。没有显示，或者显示`Cannot GET /view.html `回头检查。
> 3. 最后 Iframe 预览里 添加view.html路径 如图，后缀添加 `?src=$e_url`

现在自己的搞定了。再次感谢您！朋友的alist容器是在局域网使用，没有公网ip,安装在win7电脑上，用的win64版本的alist，所以alist主页访问时是http://192.168.10.30:5244，他现在 Iframe 预览里 添加我服务器的view.html路径，出现onlyoffice后提示“文件加载中”转圈很久然后出现“错误：加载失败”，是不是ip地址访问的原因，有没有方便解决的办法？

**Reply by chenjuefei** [reply]
我浅见：你需要简单的明白ONLYOFFICE的工作原理。可以参考[https://api.onlyoffice.com/zh/editors/open](https://api.onlyoffice.com/zh/editors/open)。 `alist` `view.html所在服务器` `Document Server` 之间需要能相互访问。你的服务器显然不能访问你朋友的内网地址。我自己理解，不一定对，你可以多做尝试，如果成功，欢迎分享！

**Reply by Westwall98** [reply]
> 1. 首先搭好onlyoffice/documentserver后，浏览网址 `https://你的域名:端口/web-apps/apps/api/documents/api.js` 查看能否看到内容。如果没有说明没有搭建好onlyoffice或者路径错误。 我这里域名地址访问，内网IP地址后续会有问题，建议用域名。
> 2. 编辑好view.html 唯一要修改的api.js地址，改成你自己的。浏览网址 `https://你的域名:端口/路径/view.html` 如果显示 `The "document.fileType" parameter for the config object is invalid. Please correct it.` 表示正常。没有显示，或者显示`Cannot GET /view.html `回头检查。
> 3. 最后 Iframe 预览里 添加view.html路径 如图，后缀添加 `?src=$e_url`

我按照您说的1和2都有正确显示结果，但在Alist中打开office文件会一直转“文档加载中”，然后过了很久提示“下载失败”。目前alist和onlyoffice用的都是https反向代理且不是443端口，请问下这会有影响吗？

**Reply by chenjuefei** [reply]
> 我按照您说的1和2都有正确显示结果，但在Alist中打开office文件会一直转“文档加载中”，然后过了很久提示“下载失败”。目前alist和onlyoffice用的都是https反向代理且不是443端口，请问下这会有影响吗？

反向代理应该不会有影响。如果你确定这个文档可以被下载和打开的话。检查网络环境，确定view.html和documentserver能够访问到这个文档。view.html也可以尝试换个地方。再试试

**Reply by kknight-ddark** [reply]
大佬请教下，这边编写view后访问已经可以返回The "document.fileType" parameter for the config object is invalid. Please correct it.在alist添加的是本机存储，访问时也是提示下载失败，实际alist是可以下载的，大概率是onlyoffice访问不到文件。alist部署环境是localhost，onlyoffice是部署在docker里的映射宿主机的80端口，都是http访问...

**Reply by kknight-ddark** [reply]
onlyoffice-run的时候也将需要在线编辑的文件地址映射到容器内了，现在在onlyoffice的例子里上传打开是没问题的。可在alist里还是下载错误。

**Reply by tingyue-wu** [reply]
> onlyoffice-run的时候也将需要在线编辑的文件地址映射到容器内了，现在在onlyoffice的例子里上传打开是没问题的。可在alist里还是下载错误。

容器和容器之间是不是不相通？我也是和你一样的问题。不过我不需要编辑，就没管。你试试把localhost改成0.0.0.0试试。

**Reply by tingyue-wu** [reply]
> > 

搞定了，style="height:90vh"   还有末尾"height": "110%",就可以了，滚动条基本不动。如果两个都设置100%反而需要滚动条移动。

**Reply by Masterlovsky** [reply]
> > >
> 
> 搞定了，style="height:90vh" 还有末尾"height": "110%",就可以了，滚动条基本不动。如果两个都设置100%反而需要滚动条移动。

请问下载失败问题是怎么解决的呢，style=“height:90vh”这句加在哪里啊？

**Reply by Masterlovsky** [reply]
> > >
> 
> 搞定了，style="height:90vh" 还有末尾"height": "110%",就可以了，滚动条基本不动。如果两个都设置100%反而需要滚动条移动。

我也搞定了！不是这个比例的问题，下载错误是因为onlyoffice跨域请求导致的（我的alist和onlyoffice都是跑在docker里的）
解决方法：
进到onlyoffice的容器中，vim /etc/onlyoffice/documentserver/default.json，找到“rejectUnauthorized”改成false，重启容器。就行了！

**Reply by zhushen12580** [reply]
> 2\. The "document.fileType" parameter for the config object is invalid. Please correct it.

大佬好
我第二步返回了“The "document.fileType" parameter for the config object is invalid. Please correct it.” 但是做了第三步，还是只弹出这个提示。
查了包，发现是api解析的是阿里云盘的直链而非我网站文件地址
`正确的应该是这样：  http://IP:端口/web-apps/apps/api/documents/view.html?src=http://yunpan.zhushen12580.top/d/%E5%85%AC%E7%9B%98/%E4%B8%B4%E6%97%B6/%E5%85%B3%E9%94%AE%E8%AF%8D%E8%A1%A8%E5%A4%A7%E8%AF%8D.xlsx
错误的是：http://IP:端口/web-apps/apps/api/documents/view.html?src=https://cn-beijing-data.aliyundrive.net/CIsgHldL%2F44508144%2F633e759723e19f50e0fd494cbd5b79e1bd9c5b21%2F633e75977f5fc5219efe48a7b8e342a18c24bb34?di=bj29&dr=44508144&f=6409a7d2ad891990f2ff4369915bf13bbf4f778c&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%27%25E5%2585%25B3%25E9%2594%25AE%25E8%25AF%258D%25E8%25A1%25A8%25E5%25A4%25A7%25E8%25AF%258D.xlsx&security-token=CAIS%2BgF1q6Ft5B2yfSjIr5bac%2Ff3u5542YyYZkDWj0Fsfewbi6jSiDz2IHFPeHJrBeAYt%2FoxmW1X5vwSlq5rR4QAXlDfNRzsbGrSqVHPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLukMWr%2Fwq3PIdp2ma447NWQlLnzyCMvvJ9OVDFyN0aKEnH7J%2Bq%2FzxhTPrMnpkSlacGoABgDT%2BliwCsl3y0O7aWuoNrEjD1DlD01GaSJgmKQP6ReoGAu1v34jkew3TDJqrhGTlVruthRC7Aef7tTweWGfJHnDuOulO33adqeXlCTEjUtxrQDrFdUJcawR7XwCBirh%2FdY9r5Nu%2FZhqhCt5Q51ITVDSyZtLnEC8Z8aI42KCGzLQ%3D&u=254808010963075794&x-oss-access-key-id=STS.NUo8MCVAYnNsdfgkA9qc7dmyj&x-oss-expires=1678461324&x-oss-signature=9AJ%2BKiEAXO0R8RXe640jycJcPrrDVqCyqmgxb%2FJURWg%3D&x-oss-signature-version=OSS2
`
alist:v3.13.2
onlyoffice:windows社区版

![image](https://user-images.githubusercontent.com/60245376/224246479-f4413d9a-1636-4fd7-8ac3-e819cdc97d96.png)

**Reply by zhushen12580** [reply]
> > 2. The "document.fileType" parameter for the config object is invalid. Please correct it.
> 
> 大佬好 我第二步返回了“The "document.fileType" parameter for the config object is invalid. Please correct it.” 但是做了第三步，还是只弹出这个提示。 查了包，发现是api解析的是阿里云盘的直链而非我网站文件地址 `正确的应该是这样： http://IP:端口/web-apps/apps/api/documents/view.html?src=http://yunpan.zhushen12580.top/d/%E5%85%AC%E7%9B%98/%E4%B8%B4%E6%97%B6/%E5%85%B3%E9%94%AE%E8%AF%8D%E8%A1%A8%E5%A4%A7%E8%AF%8D.xlsx 错误的是：http://IP:端口/web-apps/apps/api/documents/view.html?src=https://cn-beijing-data.aliyundrive.net/CIsgHldL%2F44508144%2F633e759723e19f50e0fd494cbd5b79e1bd9c5b21%2F633e75977f5fc5219efe48a7b8e342a18c24bb34?di=bj29&dr=44508144&f=6409a7d2ad891990f2ff4369915bf13bbf4f778c&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%27%25E5%2585%25B3%25E9%2594%25AE%25E8%25AF%258D%25E8%25A1%25A8%25E5%25A4%25A7%25E8%25AF%258D.xlsx&security-token=CAIS%2BgF1q6Ft5B2yfSjIr5bac%2Ff3u5542YyYZkDWj0Fsfewbi6jSiDz2IHFPeHJrBeAYt%2FoxmW1X5vwSlq5rR4QAXlDfNRzsbGrSqVHPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLukMWr%2Fwq3PIdp2ma447NWQlLnzyCMvvJ9OVDFyN0aKEnH7J%2Bq%2FzxhTPrMnpkSlacGoABgDT%2BliwCsl3y0O7aWuoNrEjD1DlD01GaSJgmKQP6ReoGAu1v34jkew3TDJqrhGTlVruthRC7Aef7tTweWGfJHnDuOulO33adqeXlCTEjUtxrQDrFdUJcawR7XwCBirh%2FdY9r5Nu%2FZhqhCt5Q51ITVDSyZtLnEC8Z8aI42KCGzLQ%3D&u=254808010963075794&x-oss-access-key-id=STS.NUo8MCVAYnNsdfgkA9qc7dmyj&x-oss-expires=1678461324&x-oss-signature=9AJ%2BKiEAXO0R8RXe640jycJcPrrDVqCyqmgxb%2FJURWg%3D&x-oss-signature-version=OSS2 ` alist:v3.13.2 onlyoffice:windows社区版
> 
> ![image](https://user-images.githubusercontent.com/60245376/224246479-f4413d9a-1636-4fd7-8ac3-e819cdc97d96.png)

问题发现了iframe的配置应该是$durl而非$e_url

**Reply by zhushen12580** [reply]
> 问题发现了iframe的配置应该是$durl而非$e_url

您好 请问windows社区版怎么设置关闭jwt？

**Reply by zhushen12580** [reply]
> 因为没有view.html 需要你自己编写
> 
> ```
> <!DOCTYPE html>
> <html lang="en">
> <head>
>     <meta charset="UTF-8">
>     <title>OnlyOffice Viewer</title>
> </head>
>  
> <body>
>     <div id="placeholder"></div>
>     <script type="text/javascript" src="https://documentserver/web-apps/apps/api/documents/api.js"></script>
>     <script>
>         function getQueryParamValue(name) {
>             const searchParams = new URLSearchParams(window.location.search);
>             return searchParams.get(name);
>         }
>         
>         const url = decodeURIComponent(getQueryParamValue("src"));
>         const fileName = url.substring(url.lastIndexOf('/') + 1, url.lastIndexOf('?') != -1 ? url.lastIndexOf('?') : url.length);
>         const fileExtension = fileName.split('.').pop();
>         const docEditor = new DocsAPI.DocEditor("placeholder", {
>             "document": {
>                 "fileType": fileExtension,
>                 "permissions": {
>                     "edit": false,
>                     "comment": true,
>                     "download": true,
>                     "print": true,
>                     "fillForms": true,
>                 },
>                 "title": fileName,
>                 "url": url,
>             },
>             "editorConfig": {
>                 "lang": "zh-CN",
>                 "mode": "view",
>             },
>             "height": "1080px",
>             "type": "desktop",
>         });
>     </script>
> </body>
> </html>
> ```
> 
> ![image](https://user-images.githubusercontent.com/35056864/219602826-fe2a0e9f-3a4b-4e9e-b442-0ca2324b2977.png)
> 
> 另：onlyoffice服务器，需要关闭JWT不然一直报“文档安全令牌未正确形成”。如果有大神研究下不用关闭也能用的话，那最好！如有BUG。欢迎指正！
> 
> ![image](https://user-images.githubusercontent.com/35056864/219597958-6ba1822a-a082-488a-9fc5-a754fc96f302.png)

大佬 编辑的html可以发一下吗 谢谢

**Reply by homuraLan** [reply]
> 2\. The "document.fileType" parameter for the config object is invalid. Please correct it.

大佬请问一下我这边已经配置好这个了但是alist加上网址后没选项咋回事，
![image](https://user-images.githubusercontent.com/96775034/235313066-2d6a8c52-a7f5-41bb-898c-30838379a9bd.png)

![image](https://user-images.githubusercontent.com/96775034/235313003-c392d6e1-62f7-497e-9942-02b40ae77b6f.png)
我是用nginx代理了html直接访问网址就是这个反馈
后缀加了文件名也没用
确定宿主机上权限是够的，用的群晖应用版不知道是不是这个问题
我排查出什么原因了，一个配置文件的奇葩bug,在最后一个不能加逗号

**Reply by CrazyJasonwell** [reply]
请问大佬view.html文件存放在哪个目录下

**Reply by CrazyJasonwell** [reply]
请问style=“height:90vh” 加到哪里？

**Reply by homuraLan** [reply]
https://github.com/Sincejunly/officeaanglist.git

**Reply by Arnozzc87** [reply]
请问下大佬们，在当前页面中是不显示的，全屏就可以，这是哪里的问题呀？/view.html?src=$durl，我的后缀是durl，不是e_url
![image](https://github.com/alist-org/alist/assets/23256026/c5e4556a-0796-4e38-9684-6eac1dd62987)
![image](https://github.com/alist-org/alist/assets/23256026/c67e9b04-6b74-4b32-8833-6f7cd3b22ae6)

**Reply by junjiangsong** [reply]
![image](https://github.com/alist-org/alist/assets/113816114/992de97d-e985-4529-9438-f3a74e618d95)
大佬们，我解决了下载失败后又提示了这个错，是什么原因呢？alist和onlyoffice都是docker部署的

**Reply by monobo1989** [reply]
> ![image](https://private-user-images.githubusercontent.com/113816114/281326661-992de97d-e985-4529-9438-f3a74e618d95.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0ODExNDAsIm5iZiI6MTcyMDQ4MDg0MCwicGF0aCI6Ii8xMTM4MTYxMTQvMjgxMzI2NjYxLTk5MmRlOTdkLWU5ODUtNDUyOS05NDM4LWYzYTc0ZTYxOGQ5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwOFQyMzIwNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMDc1MDZhYjAwZTU0OTVhMjFlMmMzMzc0OWE5MDZhZjE2MTQ1MTY4MzM1OTljNWI3MzM4NDNjNzM1MWE5Y2ZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NRMnQlHWUkgzKQwRTMy1zyKlDUIC4xm6i8TTLD6RaAY) 大佬们，我解决了下载失败后又提示了这个错，是什么原因呢？alist和onlyoffice都是docker部署的

我也这样，你解决了吗

**Reply by keyian666** [reply]
可以放在终端路径 /var/www/onlyoffice/documentserver/web-apps下，同時預覽接口路徑後面也要加入/web-apps

**Comment by nxliubo** [comment]
请问大佬，这个view.html放在onlyoffice的哪个路径下？有现成的命令没有？

**Reply by zhushen12580** [reply]
> 请问大佬，这个view.html放在onlyoffice的哪个路径下？有现成的命令没有？

有偿解答

**Reply by nxliubo** [reply]
可以的，你说个价儿。

**Reply by zhushen12580** [reply]
> 可以的，你说个价儿。

已经有大佬回答了
![image](https://user-images.githubusercontent.com/60245376/226299700-7fdafa39-6e78-4eba-b0d9-dbc35fa8880c.png)

**Reply by nxliubo** [reply]
谢谢大佬！

**Reply by CrazyJasonwell** [reply]
请问view.html文件存放路径是什么？

**Comment by CrazyJasonwell** [comment]
> > > >
> > 
> > 
> > 搞定了，style="height:90vh" 还有末尾"height": "110%",就可以了，滚动条基本不动。如果两个都设置100%反而需要滚动条移动。
> 
> 请问下载失败问题是怎么解决的呢，style=“height:90vh”这句加在哪里啊？

style=“height:90vh”这句加在哪里啊？

**Reply by tingyue-wu** [reply]
这句只是调整显示框大小，和“下载失败”毫无关系。下载失败是view.html没有加载成功引起的。

**Reply by Nerders** [reply]
我也想要问下这个加在哪里呀，目前表格的进度条和容器的进度条不同步，想要只保留一个。
![image](https://github.com/alist-org/alist/assets/56071530/efbf9667-1cec-494c-aebd-67970023722d)

**Reply by EWEDLCM** [reply]
> 这句话只是调整显示框大小，和“下载失败”无关。下载失败是view.html没有加载成功引起的。

你好，请问下载失败应该如何处理呢？找了官方文档没研究出问题所在。望解答，十分感谢

**Reply by yyqwe158** [reply]
> 如果是在内网搞的，可以参考这个链接https://blog.csdn.net/m0_53401243/article/details/133869439    （csdn将就一下）

**Reply by happygo0629** [reply]
> 我也想要问下这个加在哪里呀，目前表格的进度条和容器的进度条不同步，想要只保留一个。 ![图像](https://private-user-images.githubusercontent.com/56071530/329995750-efbf9667-1cec-494c-aebd-67970023722d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NDU0NDAsIm5iZiI6MTc0NDk0NTE0MCwicGF0aCI6Ii81NjA3MTUzMC8zMjk5OTU3NTAtZWZiZjk2NjctMWNlYy00OTRjLWFlYmQtNjc5NzAwMjM3MjJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDAyNTkwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRmNGNlMmRjZjQ0MjViYzZlOGY0ZWM4ZjFiZGZjYjQwZWEzNTI5NWYwYjE0OGY5NTJjM2NkZGY5ZTczOTU0N2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6NBpmgzVgmj3jNlA0sjtq-5W88FA8YZ_cTk8apOlq-c)

![image](https://github.com/user-attachments/assets/d8905209-6692-44a5-bf93-6f102de97d1b)
电脑上看PPT 都这样    手机端 PDF  PPT  等 都翻不到最后一页

**Comment by homuraLan** [comment]
https://github.com/Sincejunly/officeaanglist.git

CrazyJasonwell ***@***.***> 于2023年7月16日周日 19:05写道：

**Reply by happygo0629** [reply]
进度条的问题还多麻烦  手机上直接翻不到最后一页

**Comment by Arnozzc87** [comment]
![image](https://github.com/alist-org/alist/assets/23256026/c6881b33-b1d5-43fa-93e6-c1fa43a0f87d)
![image](https://github.com/alist-org/alist/assets/23256026/baa9acb5-a118-4e7d-9b68-4dd2d3fbe6d8)
请问一下，我在原页面是黑屏状态的，但是在全屏浏览就是正常的，这个是哪里没设置对吗?另外后缀我用的?src=$durl"

**Reply by steven0395** [reply]
我也是这种情况，请问你解决了吗？

**Reply by Twilight-lxy** [reply]
我遇到类似问题，我的情况是alist开了https，onlyoffice没有，导致相同问题，在alist里不显示，F12检查控制台报错，报错内容是不允许https，http混合加载，解决方法是正确配置onlyoffice的https。

**Comment by homuraLan** [comment]
去项目里提一下，然后发一下具体步骤，我0.3版本快写完了，

Arnozzc87 ***@***.***> 于2023年8月22日周二 10:47写道：

**Comment by quhanj** [comment]
怎么不关闭jwt而使用啊？能不能把jwt设置起来，让alist传过去

**Reply by Twilight-lxy** [reply]
如果是docker部署的，持久化配置文件目录，改配置文件里的jwt设置，关闭jwt

**Comment by qinguoluo** [comment]
我的情况是alist和onlyoffice都安装在同一个vps上，直接从1panel的应用商店里面安装的。现在alist配置遇到了点麻烦，预览不了。

我是这样配置的：
1. 创建view.html文件，保存到/home/onlyoffice目录下。onlyoffice的服务端口是40156，容器端口是80。
src="http://vps的IP:40156/web-apps/apps/api/documents/api.js"

2. 配置openresty，新建配置文件onlyoffice.conf。

server {
    listen 40157;
    server_name  vps的IP; #这里不清楚怎么填。
 
    location / {
       root /home/onlyoffice;  
       index view.html;
    }
}

3.  alist预览配置
"onlyoffice":"http://vps的IP:40157/view.html?src=$e_url"

这样配置的结果是预览不了。我不知道哪里不对，请大佬们帮忙看看，如能指点一下，在下不胜感激。

**Comment by xiaoyaofengyuxing** [comment]
大佬，不会操作啊，求教怎么设置？？？

**Comment by Meiqiuclaw** [comment]
我也搞通了，版本是最新的8.3，内网改外网不仅要改docker的配置，如果是同一个浏览器访问还要按F12看看网页是不是被缓存了导致地址没更新。
JWT可以在docker启动前加个环境变量JWT_ENABLED=false，启动后再去docker容器内部/etc/onlyoffice/documentserver/default.json​中把"allowPrivateIPAddress"和"allowMetaIPAddress"改成true，"rejectUnauthorized"改为false 参考[这个](https://blog.pwolf.site/2023/09/15/%E5%B1%80%E5%9F%9F%E7%BD%91%E5%86%85%E9%83%A8%E7%BD%B2alist%E5%92%8Conlyoffice/)和[这个](https://blog.csdn.net/g401946949/article/details/144376873)，顺便改大"maxDownloadBytes"、"uncompressed"解除文件大小限制，参考[这个](https://www.ittel.cn/archives/11169.html#Onlyoffice_%E9%AD%94%E6%94%B9Docker%E9%95%9C%E5%83%8F%E4%B8%8B%E8%BD%BD)，最后记得重启DOCKER

**Comment by foxdodo** [comment]
发现一个问题，这个方法搭建的onlyoffice在预览alist中的文档时，如果文档路径带有emoji则会返回401，有大佬知道怎么修复吗？

**Reply by Nerders** [reply]
> 发现一个问题，这个方法搭建的onlyoffice在预览alist中的文档时，如果文档路径带有emoji则会返回401，有大佬知道怎么修复吗？

新电脑搭建出现的文档路径带emoji导致401无法预览，同样Alist+OnlyOffice版本之前搭建的平台正常，有大佬帮忙解决么

**Comment by zwx986987** [comment]
https://blog.csdn.net/qq385105501/article/details/160216917?sharetype=blogdetail&sharerId=160216917&sharerefer=PC&sharesource=qq385105501&spm=1011.2480.3001.8118
我解决了，写了一篇blog
